ostd/src/arch/x86/iommu/registers/mod.rs

```diff
@@ -0,0 +1,152 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Registers and their definition used by IOMMU.
+
+#![allow(unused_imports)]
+#![allow(dead_code)]
+#![allow(unused_variables)]
```

Comment:
These three module-level attributes are strongly discouraged when introducing new code. `#[allow(dead_code)]` can be used for some ABI or hardware related definitions, but don't make it module-wide. Please fix it according to our [guidelines](https://asterinas.github.io/book/to-contribute/style-guidelines/rust-guidelines.html).